catalog/spatial/geocoding/spatial-geocoding-admin-module/src/main/webapp/js/view/Ingest.view.js

```diff
@@ -0,0 +1,241 @@
+                    url: '/services/content',
+                    paramName: 'file',
+                    dataType: 'json',
+                    maxFileSize: 5000000,
+                    add: function(e, data){
+                        var that = this;
+                        var model = view.buildModelFromFileData(data);
+                        model.fileuploadObject = {
+                            ref: that,
+                            e: e,
+                            data: data
+                        };
+
+                        // Allow only one item at a time to be uploaded
+                        var one = view.collection.findWhere({state : "start"});
+                        view.collection.remove(one);
+                        // Allow no duplicate entries
+                        var duplicates = view.collection.findWhere({name : model.attributes.name});
+
+                        if(typeof duplicates === "undefined") {
+                            view.collection.add(model);
+                        }
+                    },
+                    done: function(e, data){
+                        var attrs = {};
+                        attrs.name = data.files[0].name;
+                        var model = view.collection.findWhere(attrs);
+
+
+                        var contentId = data.result.properties.title;
+                        var contentFolder = data.result.properties["resource-uri"].split(":")[1];
+                        model.set("progress", 50);
+                        view.post_local_file(contentFolder, contentId, model);
+                    },
+                    fail: function(e, data){
+                        var attrs = {};
+                        attrs.name = data.files[0].name;
+                        var model = view.collection.findWhere(attrs);
+                        if (data.textStatus === 'abort') {
+                            model.set('state', 'cancelled');
+                        } else {
+                            model.set({
+                                state: 'failed',
+                                error: data.errorThrown
+                            });
+                        }
+                        view.setStartButton();
+                    },
+                    progress: function(e, data){
+                        var progress = parseInt(data.loaded / data.total * 50, 10);
+                        var attrs = {};
+                        attrs.name = data.files[0].name;
+                        var model = view.collection.findWhere(attrs);
+                        if(model) {
+                            model.set('progress', progress);
+                        }
+                    }
+                });
+                view.$('.fileupload').fileupload('option', {
+                    dropZone: view.$el
+                });
+            },
+            buildModelFromFileData: function(data){
+                var name = data.files[0].name.toLowerCase();
```

Comment:
This looks like an unused variable to me. We should be pushing this sort of logic down to a model. We could simply create a model from "data" and let the model parse that object however it wants to.